meson.build

```diff
@@ -401,6 +401,12 @@ if cc.get_id() == 'clang'
         ]
 endif
 
+if get_option('mode') == 'developer'
+        possible_common_cc_flags += '-ftrivial-auto-var-init=pattern'
```

Comment:
Looks like it messes up `MSan` and `valgrind`. Without it they can detect an unitialized variable:
```sh
$ clang -fsanitize=memory  main.c
$ ./a.out
==50642==WARNING: MemorySanitizer: use-of-uninitialized-value
    #0 0x49d724 in main (/home/vagrant/systemd/a.out+0x49d724)
    #1 0x7f234c472f1f in __libc_start_call_main (/lib64/libc.so.6+0x40f1f)
    #2 0x7f234c472fcf in __libc_start_main@GLIBC_2.2.5 (/lib64/libc.so.6+0x40fcf)
    #3 0x41c314 in _start (/home/vagrant/systemd/a.out+0x41c314)

SUMMARY: MemorySanitizer: use-of-uninitialized-value (/home/vagrant/systemd/a.out+0x49d724) in main
Exiting

$ clang  main.c
$ DEBUGINFOD_URLS= valgrind ./a.out
==50651== Memcheck, a memory error detector
==50651== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==50651== Using Valgrind-3.18.1 and LibVEX; rerun with -h for copyright info
==50651== Command: ./a.out
==50651==
==50651== Conditional jump or move depends on uninitialised value(s)
==50651==    at 0x40111F: main (in /home/vagrant/systemd/a.out)
```
With `-ftrivial-auto-var-init=pattern` they miss it completely
```sh
$ clang -fsanitize=memory -ftrivial-auto-var-init=pattern main.c
$ ./a.out

$ clang -ftrivial-auto-var-init=pattern main.c
$ DEBUGINFOD_URLS= valgrind ./a.out
...
==50675== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
```

Reply:
I guess it's working as intended? :D

I changed the logic a little. I hope that's enough, short of adding another meson build option.

Reply:
> I guess it's working as intended? :D

If it had crashed it would have worked. My understanding is that `-ftrivial-auto-var-init=pattern` can't be used to detect unitialized variables reliably so I don't think it should be used when `systemd` is built with MSan or run under Valgrind.

Reply:
It's "working as intended" because the flag makes the compiler initialize those vars, removing the undefined behavior altogether.

Reply:
The idea behind `-ftrivial-auto-var-init=pattern` as far as I understand is to initialize variables using values that most likely trigger crashes down the road. compilers aren't smart enough to inject values that trigger crashes all the time.

Reply:
Anyway, I get what you're saying. Sorry.

Reply:
Well, it depends on the type of var. If it was a pointer, it should have crashed.

Reply:
Agreed. I think it there had been a pointer there it had crashed.

FWIW With clang it appears `-ftrivial-auto-var-init=pattern` isn't always usable: https://reviews.llvm.org/D68115. That probably can explain why the tests where unions were initialized explicitly with `{}` triggered the assertions.